src/async_wrap.cc

```diff
@@ -410,15 +410,21 @@ void AsyncWrap::PopAsyncIds(const FunctionCallbackInfo<Value>& args) {
 void AsyncWrap::AsyncReset(const FunctionCallbackInfo<Value>& args) {
   AsyncWrap* wrap;
   ASSIGN_OR_RETURN_UNWRAP(&wrap, args.Holder());
-  double execution_async_id = args[0]->IsNumber() ? args[0]->NumberValue() : -1;
+  double execution_async_id =
+      args[0]->IsNumber()
+          ? args[0]
+                ->NumberValue(args.GetIsolate()->GetCurrentContext())
+                .ToChecked()
+          : -1;
   wrap->AsyncReset(execution_async_id);
 }
 
 
 void AsyncWrap::QueueDestroyAsyncId(const FunctionCallbackInfo<Value>& args) {
   CHECK(args[0]->IsNumber());
   AsyncWrap::EmitDestroy(
-      Environment::GetCurrent(args), args[0]->NumberValue());
+      Environment::GetCurrent(args),
+      args[0]->NumberValue(args.GetIsolate()->GetCurrentContext()).ToChecked());
```

Comment:
Umm, dunno. There's literally a `CHECK` above, so I thought `ToChecked` would work here. I mean, it might still crash, sure, but nowhere it didn't crash earlier.

Reply:
Oh, right, I missed that – in that case, we don’t need `NumberValue()` either and can cast via `.As<Number>()`, right?

Reply:
:P You're right. Updating the PR.